### PR TITLE
Create basic idea of menu.

### PR DESCRIPTION
**Pull request by SamBumgardner**
Not quite complete yet. Accepts input, emits signals based on directions selected, and does some visual highlighting for the selected option.

Would be nice to have the dialogue box type stuff too, but we'll see where we want to draw the line for this branch.

### CODE REVIEW

File: godot/menu/Menu.gd [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+			"right " + randomText, "down " + randomText])
+	
+	var direction = _getInputDirection();
+	if _acceptingInput && direction != Direction.NONE:
+		if OS.is_debug_build():
+			print("Menu direction pressed: ", direction)
+		
+		_menuOptions[direction].get_node("Arrow").texture = _optionHighlightTexture
+		emit_signal("menuDirectionSelected", direction)
+		_acceptingInput = false
```
Comment (NBumgardner) [inline godot/menu/Menu.gd]: **Suggestion:** Maybe encapsulate this in some kind of `chooseDirection(direction)` function? That would also help with the long-line complaint in the IDE, and let `_process` not worry about the implementation.

**Review by NBumgardner — COMMENTED** [review]
Looks good, one suggestion on `_process`.

**Review by NBumgardner — DISMISSED** [review]

File: godot/menu/Menu.gd (hunk 1 above)
Comment (SamBumgardner) [reply godot/menu/Menu.gd]: it is done

**Comment by NBumgardner** [comment]
Looks good, I can still see your scene with the Play Scene (F6) option, after opening `res://menu/Menu.tscn`.

**Review by NBumgardner — APPROVED** [review]

**Pull request merged**